core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -235,6 +236,7 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
 
     private void cleanup() {
         if (!unfinishedRequests.isEmpty()) {
+            isCleaning = true;
```

Comment:
nit; what do you think of clearing `unfinishedRequests` once the iteration is done in this method?